bundles/aws-bundle/src/main/java/org/apache/gravitino/s3/credential/S3TokenProvider.java

```diff
@@ -0,0 +1,241 @@
+import org.apache.gravitino.credential.CredentialProvider;
+import org.apache.gravitino.credential.PathBasedCredentialContext;
+import org.apache.gravitino.credential.S3TokenCredential;
+import org.apache.gravitino.credential.config.S3CredentialConfig;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.policybuilder.iam.IamConditionOperator;
+import software.amazon.awssdk.policybuilder.iam.IamEffect;
+import software.amazon.awssdk.policybuilder.iam.IamPolicy;
+import software.amazon.awssdk.policybuilder.iam.IamResource;
+import software.amazon.awssdk.policybuilder.iam.IamStatement;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+import software.amazon.awssdk.services.sts.model.AssumeRoleResponse;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+/** Generates S3 token to access S3 data. */
+public class S3TokenProvider implements CredentialProvider {
+  private StsClient stsClient;
+  private String roleArn;
+  private String externalID;
+  private int tokenExpireSecs;
+
+  @Override
+  public void initialize(Map<String, String> properties) {
+    S3CredentialConfig s3CredentialConfig = new S3CredentialConfig(properties);
+    this.roleArn = s3CredentialConfig.s3RoleArn();
+    this.externalID = s3CredentialConfig.externalID();
+    this.tokenExpireSecs = s3CredentialConfig.tokenExpireSecs();
+    this.stsClient = createStsClient(s3CredentialConfig);
+  }
+
+  @Override
+  public void close() {
+    if (stsClient != null) {
+      stsClient.close();
+    }
+  }
+
+  @Override
+  public String credentialType() {
+    return S3TokenCredential.S3_TOKEN_CREDENTIAL_TYPE;
+  }
+
+  @Override
+  public Credential getCredential(CredentialContext context) {
+    if (!(context instanceof PathBasedCredentialContext)) {
+      return null;
+    }
+    PathBasedCredentialContext pathBasedCredentialContext = (PathBasedCredentialContext) context;
+    Credentials s3Token =
+        getS3Token(
+            roleArn,
+            pathBasedCredentialContext.getReadPaths(),
+            pathBasedCredentialContext.getWritePaths(),
+            pathBasedCredentialContext.getUserName());
+    return new S3TokenCredential(
+        s3Token.accessKeyId(),
+        s3Token.secretAccessKey(),
+        s3Token.sessionToken(),
+        s3Token.expiration().toEpochMilli());
+  }
+
+  private StsClient createStsClient(S3CredentialConfig s3CredentialConfig) {
+    AwsCredentialsProvider credentialsProvider =
+        StaticCredentialsProvider.create(
+            AwsBasicCredentials.create(
+                s3CredentialConfig.accessKeyID(), s3CredentialConfig.secretAccessKey()));
+    String region = s3CredentialConfig.region();
```

Comment:
Builder builder = StsClient.builder()
    .credentialsProvider(credentialsProvider);
if (StringUtils.isNotBlank(region)) {
      builder.region(Region.of(region));  
}
return builder.build();

Reply:
updated